packages/region/src/services/region-module.ts

```diff
@@ -0,0 +1,230 @@
+import {
+  Context,
+  CreateRegionDTO,
+  DAL,
+  InternalModuleDeclaration,
+  IRegionModuleService,
+  ModuleJoinerConfig,
+  ModulesSdkTypes,
+  RegionCountryDTO,
+  RegionCurrencyDTO,
+  RegionDTO,
+  UpdateRegionDTO,
+} from "@medusajs/types"
+import {
+  InjectManager,
+  InjectTransactionManager,
+  MedusaContext,
+  MedusaError,
+  ModulesSdkUtils,
+  promiseAll,
+} from "@medusajs/utils"
+
+import { Country, Currency, Region } from "@models"
+
+import { DefaultsUtils } from "@medusajs/utils"
+import { CreateCountryDTO, CreateCurrencyDTO } from "@types"
+import { entityNameToLinkableKeysMap, joinerConfig } from "../joiner-config"
+
+type InjectedDependencies = {
+  baseRepository: DAL.RepositoryService
+  regionService: ModulesSdkTypes.InternalModuleService<any>
+  countryService: ModulesSdkTypes.InternalModuleService<any>
+  currencyService: ModulesSdkTypes.InternalModuleService<any>
+}
+
+const generateMethodForModels = [Country, Currency]
+
+export default class RegionModuleService<
+    TRegion extends Region = Region,
+    TCountry extends Country = Country,
+    TCurrency extends Currency = Currency
+  >
+  extends ModulesSdkUtils.abstractModuleServiceFactory<
+    InjectedDependencies,
+    RegionDTO,
+    {
+      Country: {
+        dto: RegionCountryDTO
+        singular: "Country"
+        plural: "Countries"
```

Comment:
todo: this can be removed, it will be infered, this option is more when wehave a tricky word or we want to force the values